Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/AssetCreator.h

```diff
@@ -153,15 +153,19 @@ namespace AZ
         void AssetCreator<AssetDataT>::ReportError([[maybe_unused]] const char* format, [[maybe_unused]] Args... args)
         {
             ++m_errorCount;
+#if defined(AZ_ENABLE_TRACING) // disabling since it requires argument expansion in this context
             AZ_Error(m_assetClassName, false, format, args...);
+#endif
         }
 
         template<typename AssetDataT>
         template<typename ... Args>
         void AssetCreator<AssetDataT>::ReportWarning([[maybe_unused]] const char* format, [[maybe_unused]] Args... args)
         {
             ++m_warningCount;
+#if defined(AZ_ENABLE_TRACING) // disabling since it requires argument expansion in this context
             AZ_Warning(m_assetClassName, false, format, args...);
+#endif
```

Comment:
I had to workaround these because the template expansion of these arguments were not playing nice with the macro. The last argument ended up expanded as:
```
(void)args...
```

Reply:
That is because variadic templates arguments are expanded by the compiler, not the preprocessor.

Reply:
yeah, tried to redirect it to a function that had `[[maybe_unused]]` in the parameters (provided by @burelc-amzn) but the function does not get compiled out... since it was these two places, didnt want to waste time on it.